### PR TITLE
Add new cause of death form inputs in Farajaland from SPC implementation

### PR DESCRIPTION
**Pull request by pankaj-pant**
## Description

Issue - https://github.com/opencrvs/opencrvs-core/issues/12231

## Checklist

- [ ] I have linked the correct Github issue under "Development"
- [ ] I have tested the changes locally, and written appropriate tests
- [ ] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [ ] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: infrastructure/postgres/setup-reference-data.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+# Configuration
+: "${POSTGRES_HOST:=localhost}"
+: "${POSTGRES_PORT:=5432}"
+: "${POSTGRES_PASSWORD:?Must set POSTGRES_PASSWORD}"
+: "${POSTGRES_USER:?Must set POSTGRES_USER}"
+: "${REFERENCE_DATA_POSTGRES_PASSWORD:?Must set REFERENCE_DATA_POSTGRES_PASSWORD}"
+: "${REFERENCE_DATA_POSTGRES_USER:?Must set REFERENCE_DATA_POSTGRES_USER}"
+: "${REFERENCE_DATA_EDITOR_PASSWORD:?Must set REFERENCE_DATA_EDITOR_PASSWORD}"
+: "${REFERENCE_DATA_EDITOR_USER:?Must set REFERENCE_DATA_EDITOR_USER}"
+: "${KEEP_ALIVE_SECONDS:=0}" # Prevent Swarm from marking this task as failed due to early exit
```
Comment (adskyiproger) [inline infrastructure/postgres/setup-reference-data.sh:17]: Don't forget to make this conditional:

```suggestion
: "${POSTGRES_PASSWORD:?Must set POSTGRES_PASSWORD}"
: "${POSTGRES_USER:?Must set POSTGRES_USER}"

if [[ -z "$REFERENCE_DATA_POSTGRES_PASSWORD" ||
      -z "$REFERENCE_DATA_POSTGRES_USER" ||
      -z "$REFERENCE_DATA_EDITOR_PASSWORD" ||
      -z "$REFERENCE_DATA_EDITOR_USER" ]]; then
  echo "Required variables are not set, skipping..."
  exit 0
fi

: "${KEEP_ALIVE_SECONDS:=0}" # Prevent Swarm from marking this task as failed due to early exit
```

File: infrastructure/docker-compose.e2e-deploy.yml [hunk 2]
```diff
@@ -96,3 +96,33 @@ services:
       - --accesslog=true
       - --accesslog.format=json
       - --ping=true
+
```
Comment (pankaj-pant) [inline infrastructure/docker-compose.e2e-deploy.yml:99]: Testing e2e deployment in docker swarm mode

File: src/events/death/forms/pages/eventDetails.ts [hunk 3]
```diff
@@ -203,98 +201,6 @@ export const eventDetails = defineFormPage({
         }
       ]
     },
-    {
```
Comment (pankaj-pant) [inline src/events/death/forms/pages/eventDetails.ts:206]: Changes have been requested in this ticket - https://github.com/opencrvs/opencrvs-core/issues/12231

File: src/data-seeding/roles/roles.ts [hunk 4]
```diff
@@ -33,7 +33,11 @@ export const roles: Role[] = [
       { type: 'record.request-correction', options: { registeredIn: 'administrativeArea' } },
       { type: 'record.custom-action', options: { event: ['birth'], customActionTypes: ['VALIDATE_DECLARATION', 'ESCALATE'] } },
       { type: 'record.custom-action', options: { event: ['birth'], customActionTypes: ['ISSUE_CERTIFIED_COPY', 'ISSUE_VERIFIABLE_CREDENTIAL'] } },
-      { type: 'record.custom-action', options: { event: ['death'], customActionTypes: ['VALIDATE_DECLARATION'] } }
+      { type: 'record.custom-action', options: { event: ['death'], customActionTypes: ['VALIDATE_DECLARATION'] } },
+      {
+        type: 'dashboard.view',
+        options: { ids: ['registrations', 'completeness', 'registry'] }
+      }
```
Comment (cibelius) [inline src/data-seeding/roles/roles.ts:40]: should these be added to users on opencrvs-countryconfig as well?

File: src/index.ts [hunk 5]
```diff
@@ -546,6 +548,17 @@ export async function createServer() {
...
+  server.route({
+    method: 'GET',
+    path: '/causes-of-death',
+    handler: causeOfDeathSearchHandler,
+    options: {
+      auth: false,
+      tags: ['api', 'search'],
+      description: 'Fuzzy search codes with source-based priority'
+    }
+  })
```
Comment (cibelius) [inline src/index.ts:560]: I think a comment for this handler would also be useful 👍

**Review by cibelius — APPROVED** [review]

**Comment by naftis** [comment]
@pankaj-pant I think locally we need a similar setup script as for setup-analytics/start: https://github.com/opencrvs/opencrvs-farajaland/blob/9946e2348805486fe40c0b57dcd3d3a1d5bdcd2c/package.json#L27
